### PR TITLE
feat(auth): redirect to `targetPath` after login

### LINKED ISSUE #1093: feat(auth): post-login redirect to initial url
Deeplinks to traces are a popular feature. Currently if a user is unauthenticated, the deep link does not work as expected as it is not persisted when redirecting an unauthed user to the login page (see layout.tsx).

Steps

- [ ] Add target url to auth page as url param in redirect (layout.tsx)
- [ ] Persist target url while signing in (managed by AuthJS/NextAuth)
- [ ] Add e2e test for this behavior (playwright)

### PR DESCRIPTION
**Pull request by InosRahul**
## What does this PR do?

Fixes  https://github.com/langfuse/langfuse/issues/1093

 Loom Video: https://www.loom.com/share/a5da2b975d3c42f7a2c7e63cdcc102b6?sid=a5a83a36-3fa3-4f47-a242-b68335ca96bb

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: web/src/components/layouts/layout.tsx [hunk 1]
```diff
@@ -136,31 +136,43 @@ export default function Layout(props: PropsWithChildren) {
...
+      void router.replace(targetUrl);
+    } else {
+      void router.replace("/");
+    }
     return <Spinner message="Redirecting" />;
   }
 
   const hideNavigation =
     session.status === "unauthenticated" ||
+    projects.length === 0 ||
```
Comment (marcklingen) [inline web/src/components/layouts/layout.tsx:175]: this was removed in a recent change to main, please remove

**Review by marcklingen — COMMENTED** [review]
thanks for the contribution, this looks good mostly, can you remove this line?

File: web/src/components/layouts/layout.tsx (hunk 1 above)
Comment (InosRahul) [reply web/src/components/layouts/layout.tsx]: Done.

Comment (InosRahul) [reply web/src/components/layouts/layout.tsx]: @marcklingen is this okay or does it need more changes? Also I noticed the tests CI tests are failing for forked repositories, is this intended behaviour or do I need to change some config?

File: web/src/components/layouts/layout.tsx [hunk 2]
```diff
@@ -144,18 +144,25 @@ export default function Layout(props: PropsWithChildren) {
   if (
     session.status === "unauthenticated" &&
     !unauthenticatedPaths.includes(router.pathname) &&
-    !publishablePaths.includes(router.pathname) &&
```
Comment (marcklingen) [inline web/src/components/layouts/layout.tsx:147]: why did you remove this line?

**Review by marcklingen — COMMENTED** [review]
made little changes (cosmetic), one last questions, otherwise seems good to go. thank you!

File: web/src/__e2e__/auth.spec.ts [hunk 3]
```diff
@@ -70,3 +70,54 @@ test("Signup input validation", async ({ page }) => {
...
+
+  await page.getByRole("button", { name: "Tracing" }).click();
+
+  await page.getByRole("link", { name: "Traces" }).click();
+
+  await page.waitForTimeout(2000);
+
+  await expect(page).toHaveURL(/.*traces/);
+
+  await page
```
Comment (marcklingen) [inline web/src/__e2e__/auth.spec.ts:93]: this test seems to fail, can you fix it?

**Review by marcklingen — CHANGES_REQUESTED** [review]
made additional changes, PR ready to merge but tests keep failing. Can you fix the end to end test?

**Comment by marcklingen** [comment]
maybe you can just try to navigate to /settings of the project without really needing to press any buttons beyond the login screen, assuming that this is the issue here

**Comment by InosRahul** [comment]
> maybe you can just try to navigate to /settings of the project without really needing to press any buttons beyond the login screen, assuming that this is the issue here

Fixed now, also cleaned up the commits.

File: web/src/components/layouts/layout.tsx (hunk 1 above)
Comment (marcklingen) [reply web/src/components/layouts/layout.tsx]: we fixed the ci test, sorry for the confusion here

**Comment by marcklingen** [comment]
Looks great, the worker CI test fails on forks, we're about to fix this to make it easier to contribute to langfuse

**Comment by marcklingen** [comment]
Thank you for your contribution!
